src/core_init_atmosphere/mpas_geotile_manager.F

```diff
@@ -0,0 +1,487 @@
+   !  function mpas_add_tile
+   !
+   !> \brief   Reads in a tile from disk and allocates it to main memory
+   !> \author  Miles A. Curry
+   !> \date    
+   !> \details 
+   !>    This function reads a datafile from disk and allocates it on the heap
+   !> /main memory. Given the latitude and longitude of a point, it calculates
+   !> which tilefile contains that points and reads in the datafile that contains
+   !> that point.
+   !
+   !-----------------------------------------------------------------------
+   subroutine mpas_add_tile(lat, lon, tile, tile_hash, geo_pool)
+
+      use mpas_c_interfacing, only : mpas_f_to_c_string
+      use iso_c_binding, only : c_char 
+
+      implicit none
+
+      interface
+         subroutine read_geogrid(fname, rarray, nx, ny, nz, isigned, endian, &
+                                 scalefactor, wordsize, status) bind(C)
+            use iso_c_binding, only : c_char, c_int, c_float, c_ptr
+            character (c_char), dimension(*), intent(in) :: fname
+            type (c_ptr), value :: rarray
+            integer (c_int), intent(in), value :: nx
+            integer (c_int), intent(in), value :: ny
+            integer (c_int), intent(in), value :: nz
+            integer (c_int), intent(in), value :: isigned
+            integer (c_int), intent(in), value :: endian
+            real (c_float), intent(in), value :: scalefactor
+            integer (c_int), intent(in), value :: wordsize
+            integer (c_int), intent(inout) :: status
+         end subroutine read_geogrid
+      end interface
+
+      ! Input values
+      real (kind=RKIND), intent(in), value :: lat, lon
+      type (tile_t), intent(out), pointer :: tile
+      type (hash_table), pointer, dimension(:,:), intent(inout) :: tile_hash
+      type (mpas_pool_type), intent(inout), pointer :: geo_pool
+
+      ! Local variables
+      integer :: x, y
+      character (len=StrKIND) :: fname
+      character (len=StrKIND, kind=c_char) :: c_fname
+      character (len=StrKIND), pointer :: topo_dir_path
+      type(c_ptr) :: tile_ptr
+
+      integer, pointer :: tile_nx, tile_ny, tile_nz, tile_bdr
+      integer (c_int) :: nx, ny, nz
+
+      integer, pointer :: wordsize
+      integer (c_int) :: c_wordsize
+
+      integer, pointer :: signed
+      integer (c_int) :: c_isigned
+      integer (c_int) :: status
+      
+      real, pointer :: scalefactor
+      real (c_float) :: c_scalefactor
+
+      integer, pointer :: endian
+      integer (c_int) :: c_endian
+   
+      tile => null()
+      call mpas_pool_get_config(geo_pool, 'tile_x', tile_nx)
+      call mpas_pool_get_config(geo_pool, 'tile_y', tile_ny)
+      call mpas_pool_get_config(geo_pool, 'tile_z', tile_nz)
+      call mpas_pool_get_config(geo_pool, 'tile_bdr', tile_bdr)
+      call mpas_pool_get_config(geo_pool, 'wordsize', wordsize)
+      call mpas_pool_get_config(geo_pool, 'signed', signed)
+      call mpas_pool_get_config(geo_pool, 'topo_dir_path', topo_dir_path)
```

Comment:
We should update `topo_dir_path` to be `geog_dir_path`

Reply:
That makes sense to me. I would also propose that `dir` and `path` are redundant, and we could call the variable something like `geog_path` ("path" sounds more Linuxy than "dir").